nbconvert/preprocessors/execute.py

```diff
@@ -494,14 +528,17 @@ def _get_timeout(self, cell):
 
         return timeout
 
-    def _handle_timeout(self):
+    def _handle_timeout(self, cell=None):
         self.log.error(
             "Timeout waiting for execute reply (%is)." % self.timeout)
         if self.interrupt_on_timeout:
             self.log.error("Interrupting kernel")
             self.km.interrupt_kernel()
         else:
-            raise TimeoutError("Cell execution timed out")
+            if cell is not None:
```

Comment:
Nice defensive coding! Maybe the classmethod should be able to handle `cell=None` instead of leaving it to the caller, though.